src/operation/get-parameters-as-json-schema.ts

```diff
@@ -220,26 +220,9 @@ export default function getParametersAsJSONSchema(
     }
 
     const components: Partial<ComponentsObject> = {
-      /**
-       * Initializing an empty `components[componentType] = {}` object within the `forEach` below
-       * is incredibly slow because each of these component types has a wide variety of shapes. So
-       * in order to not have TS compilation times that takes literally **seconds** because of a
-       * single line we're instead opting to prefill this object with some empty placeholders that
-       * we'll later remove if they didn't get used.
-       *
-       * Obviously not ideal but I'd rather have a couple lines of boilerplate nonsense than having
-       * to wait a noticeably frustrating amount of time for TS Intellisense to reload itself after
-       * you save a line in any file.
-       */
-      examples: {},
-      schemas: {},
-      responses: {},
-      parameters: {},
-      requestBodies: {},
-      headers: {},
-      securitySchemes: {},
-      links: {},
-      callbacks: {},
+      ...Object.keys(api.components)
+        .map(componentType => ({ [componentType]: {} }))
+        .reduce((prev, next) => Object.assign(prev, next), {}),
```

Comment:
the pros seem to outweigh the cons here but is this gonna muck up intellisense again?

Reply:
Seems to have had no affect on intellisense speeds. I think the original thing that fixed Intelisense was doing `Partial<ComponentsObject>` and since these things like `x-definitions` doesn't exist in the OAS types it's fine.